website/content/en/preview/concepts/nodeclasses.md

```diff
@@ -81,7 +81,7 @@ spec:
         volumeType: gp3
         iops: 10000
         encrypted: true
-        kmsKeyID: "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"
```

Comment:
Looking at the actual documentation, does the updated format work? All of the docs callout that this should be an ARN. In your testing, do both the ARN and the raw ID work or does ARN not work at all?

Reply:
Hi @jonathan-innis 
I was following the current documentation, but it was not working. Initially I thought its something with the KMS key policy, but that was fine. The EC2 instance tries to start and terminate immediately with client error as below.

`Client.InvalidKMSKey.InvalidState: The KMS key provided is in an incorrect state`

In the cloudtrail events I got error for event - GenerateDataKeyWithoutPlaintext as below
`"User: arn:aws:iam::<account-id>:role/karpenter-controller-role is not authorized to perform: kms:GenerateDataKeyWithoutPlaintext on this resource because the resource does not exist in this Region, no resource-based policies allow access, or a resource-based policy explicitly denies access",`

The launch template looked good, but while I compared it with the launch template of the node-groups, i noticed a small difference in the console. For the node-group launch template it showed only the keyID in the details, but for the karpenter launch template it showed full arn.
And I tested without arn prefix, surprisingly the instance started working fine and node also registered successfully..!
I believe this is a valid test.

Reply:
Huh, that's quite confusing. Definitely unexpected behavior, given what the documentation indicates around that parameter in the API.

Reply:
@Nadeshan I just tested the `kmsKeyID` with the ARN, and it worked as expected for me. Can you verify again and make sure the ARN isn't working for you?

Reply:
> The launch template looked good, but while I compared it with the launch template of the node-groups, i noticed a small difference in the console. For the node-group launch template it showed only the keyID in the details, but for the karpenter launch template it showed full arn.
> And I tested without arn prefix, surprisingly the instance started working fine and node also registered successfully..!

Did you change the kmsKeyID in the console or in EC2NodeClass?

Reply:
What I mentioned about aws console was the troubleshooting procedure, through which I identified changing to key id worked. 

I'm using keyid in ec2nodeclass definition for my original use case, not in the console. 

Also it's difficult to edit it console for this short living launch template na?

Reply:
Is the key in the same region as the instance you are trying to launch?

Reply:
ARN wasn't working for me, putting the key id alone did the job. Any connections with the AWS region used?

Reply:
@engedaam 
Yes key and instance in same region - Frankfurt.

Reply:
Ran into this same issue with `v0.33.0` - `Client.InvalidKMSKey.InvalidState: The KMS key provided is in an incorrect state` when specifying the KMS key's arn on ebs. The only thing that ended up working was specifying the key id and so far we haven't seen any issues